src/components/KeyboardCompatibleView/KeyboardCompatibleView.js

```diff
@@ -1,132 +1,266 @@
+      const subscription = Keyboard.addListener('keyboardDidHide', () => {
         resolve();
-      } else {
-        // Bring the channel height to its full length state.
-        Animated.timing(heightAnim, {
-          duration: keyboardDismissAnimationDuration,
-          toValue: initialHeight,
-          useNativeDriver: false,
-        }).start(resolve);
-      }
+        subscription.remove();
+      });
+
+      Keyboard.dismiss();
     });
-  }, [
-    channelHeight,
-    heightAnim,
-    initialHeight,
-    isKeyboardOpen,
-    keyboardDismissAnimationDuration,
-  ]);
-
-  const onLayout = useCallback(
-    ({
-      nativeEvent: {
-        layout: { height },
-      },
-    }) => {
-      if (!enabled) {
-        return;
-      }
-
-      // Not to set initial height again.
-      if (!initialHeight) {
-        setInitialHeight(height);
-        Animated.timing(heightAnim, {
-          duration: 10,
-          toValue: height,
-          useNativeDriver: false,
-        }).start();
-      }
-    },
-    [enabled, heightAnim, initialHeight],
-  );
-
-  if (!enabled) {
-    return (
-      <KeyboardContext.Provider
-        value={{
-          dismissKeyboard,
-        }}
-      >
-        {children}
-      </KeyboardContext.Provider>
-    );
+  };
+
+  componentDidMount() {
+    AppState.addEventListener('change', this._handleAppStateChange);
+    this.setKeyboardListeners();
+  }
+
+  componentWillUnmount() {
+    AppState.removeEventListener('change', this._handleAppStateChange);
+    this.unsetKeyboardListeners();
   }
 
-  return (
-    <Animated.View
-      onLayout={onLayout}
-      style={{
-        height: initialHeight ? heightAnim : undefined,
-      }}
-    >
-      <KeyboardContext.Provider value={{ dismissKeyboard }}>
-        <View collapsable={false} ref={rootChannelView}>
-          {children}
-        </View>
-      </KeyboardContext.Provider>
-    </Animated.View>
-  );
-};
+  render() {
+    const {
+      behavior,
+      children,
+      contentContainerStyle,
+      enabled,
+      keyboardVerticalOffset,
+      style,
+      ...props
+    } = this.props;
+    const bottomHeight = enabled ? this.state.bottom : 0;
+    switch (behavior) {
+      case 'height':
+        // eslint-disable-next-line no-case-declarations
+        let heightStyle;
+        if (this._frame != null && this.state.bottom > 0) {
+          // Note that we only apply a height change when there is keyboard present,
+          // i.e. this.state.bottom is greater than 0. If we remove that condition,
+          // this.frame.height will never go back to its original value.
+          // When height changes, we need to disable flex.
+          heightStyle = {
+            flex: 0,
+            height: this._initialFrameHeight - bottomHeight,
+          };
+        }
+        return (
+          <KeyboardContext.Provider
+            value={{
+              dismissKeyboard: this.dismissKeyboard,
+            }}
+          >
```

Comment:
don't think any of these context things are needed... in the other file changes it seems like you're removing usage of dismissKeyboard so KeyboardContext isn't doing anything anymore

Reply:
We use it in `MessageContent.js`